src/sentry/api_gateway/proxy.py

```diff
@@ -0,0 +1,51 @@
+"""
+Utilities related to proxying a request to a region silo
+"""
+
+
+from django.http import StreamingHttpResponse
+from requests import Response as ExternalResponse
+from requests import request as external_request
+from rest_framework.exceptions import NotFound
+from rest_framework.request import Request
+
+from sentry.models.organization import Organization
+
+# stream 0.5 MB at a time
+PROXY_CHUNK_SIZE = 512 * 1024
+
+
+def _parse_response(response: ExternalResponse) -> StreamingHttpResponse:
+    """
+    Convert the Responses class from requests into the drf Response
+    """
+    streamed_response = StreamingHttpResponse(
+        streaming_content=(chunk for chunk in response.iter_content(PROXY_CHUNK_SIZE)),
+        status=response.status_code,
+        content_type=response.headers.pop("Content-Type"),
+    )
+    # Add Headers to response
+    for header, value in response.headers.items():
+        streamed_response[header] = value
+    return streamed_response
+
+
+def proxy_request(request: Request, org_slug: str) -> StreamingHttpResponse:
+    """Take a django request opject and proxy it to a remote location given an org_slug"""
+    from sentry.types.region import get_region_for_organization
+
+    try:
+        org = Organization.objects.get(slug=org_slug)
+    except Organization.DoesNotExist:
+        raise NotFound(detail="Resource could not be found")
```

Comment:
This is potentially susceptible to an enumeration attack. Malicious attackers could use this to check if a specific org slug exists. 

https://www.upguard.com/blog/what-is-an-enumeration-attack

cc @mdtro

Reply:
It's why I didn't specify what resource couldn't be found. But yeah, good point. I'll change it.

Reply:
@AniketDas-Tekky Ah. I just wanted to raise awareness. I think it's probably non-trivial to patch around this, unless the same error also bubbles up from the org endpoint base class.

Reply:
Actually, technically 404 is the response you want to have in general. 404 is the obfuscated response